ci/test/04_install.sh

```diff
@@ -33,7 +33,12 @@ if [ -z "$DANGER_RUN_CI_ON_HOST" ]; then
 
   # the name isn't important, so long as we use the same UID
   LOCAL_USER=nonroot
-  ${CI_RETRY_EXE} docker pull "$CI_IMAGE_NAME_TAG"
+  DOCKER_BUILDKIT=1 ${CI_RETRY_EXE} docker build \
+      --file "${BASE_ROOT_DIR}/ci/test_imagefile" \
+      --build-arg "CI_IMAGE_NAME_TAG=${CI_IMAGE_NAME_TAG}" \
+      --build-arg "FILE_ENV=${FILE_ENV}" \
+      --tag="${CONTAINER_NAME}" \
+      "${BASE_ROOT_DIR}"
```

Comment:
Do we have good confidence that in Cirrus environments, local build caches will persist? If not, we'll hit slowdowns because pulling from registries is about as fast as it gets.